src/test/kotlin/com/example/realworldkotlinspringbootjdbc/domain/article/CommentIdTest.kt

```diff
@@ -0,0 +1,79 @@
+package com.example.realworldkotlinspringbootjdbc.domain.article
+
+import arrow.core.Validated.Invalid
+import arrow.core.Validated.Valid
+import arrow.core.invalidNel
+import com.example.realworldkotlinspringbootjdbc.domain.comment.CommentId
+import net.jqwik.api.ForAll
+import net.jqwik.api.Property
+import net.jqwik.api.constraints.IntRange
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+class CommentIdTest {
+    class NewTest {
+
+        @Property(tries = 100)
+        fun `正常系-自然数の場合`(
+            @ForAll @IntRange(min = 1) naturalNumber: Int
+        ) {
+            /**
+             * given:
+             */
+
+            /**
+             * when:
+             */
+            val actual = CommentId.new(naturalNumber)
+
+            /**
+             * then:
+             */
+            when (actual) {
+                is Invalid -> assert(false) { "原因: ${actual.value}" }
+                is Valid -> assertThat(actual.value.value).isEqualTo(naturalNumber)
+            }
+        }
+
+        @Test
+        fun `準正常系-null の場合、バリデーションエラーが戻り値`() {
+            /**
+             * given:
+             */
+            val nullInt = null
+
+            /**
+             * when:
+             */
+            val actual = CommentId.new(nullInt)
+
+            /**
+             * then:
+             */
+            val expected = CommentId.ValidationError.Required.invalidNel()
+            assertThat(actual).isEqualTo(expected)
+        }
+
+        @Property(tries = 100)
+        fun `準正常系-自然数でない場合`(
+            @ForAll @IntRange(min = 0) naturalNumber: Int
+        ) {
+            /**
+             * given:
+             * 0 以上の整数に -1 を乗算することで、自然数でない整数を表現する
+             */
+            val notNaturalNumber = -1 * naturalNumber
```

Comment:
[nits]

境界値としてInt.MIN_VALUEがテストできないので、 `min=Int.MIN_VALUE` をやるのはどうでしょう？

（注）//@のあとにアルファベットのみが続くとGithubのアカウントへのリンクになっていますが、関係ありません）

```suggestion
        @Property
        fun `準正常系-自然数でない場合`(
            @ForAll @IntRange(min = Int.MIN_VALUE, max = 0) notNaturalNumber: Int
        ) {
            /**
             * given:
             */
```

Int.MAX_VALUE = 2147483647
Int.MIN_VALUE = -2147483648

なので -1 * Int.MAX_VALUEについては1周回って+1とかにならずに大丈夫なのですが限界が
Intの負の値での境界値としては `(-1*Int.MAX_VALUE)` より1少ない Int.MIN_VALUE かなーと

蛇足

境界値周りの符号反転は気をつけて方が良くて

```
Int.MIN_VALUE * -1 = -2147483648です（負の値＊負の値=負の値になっている）
```

2147483648というInt値は存在せず、 `2147483647` の次（+1）は-2147483648なので、1周したみたいな感覚でしょうか

-2147483648...,-3,-2,-1,0,1,2,3,...2147483647,-2147483648

Reply:
こちら、以下で修正しました 🙇

https://github.com/sunakan/realworld-kotlin-springboot-jdbc/pull/171/commits/31c00d04328615adb6da237e5e88563a035bb3cf

sandbox のとき、min に負の整数を指定するとエラーがおきていたので、負の整数が使えないのではと思っていたのですが、どうやら勘違いだったみたいですね。。。

IntRange の min のデフォルトが 0 だっため、そこからも勘違いしていました（[参考](https://jqwik.net/docs/1.6.5/javadoc/net/jqwik/api/constraints/IntRange.html )）。